tests/ui/update_banner.js

```diff
@@ -0,0 +1,64 @@
+ /*
+     Test for the update banner states, using the mozApps mock to pretend
+     Marketplace app is installed/has an update available/etc.
+ */
+var helpers = require('../lib/helpers');
+var mozApps = require('../lib/mozApps');
+var _ = require('../../node_modules/underscore');
+
+
+casper.test.begin('Test manifest_url', {
+    test: function(test) {
+        helpers.startCasper();
+        helpers.waitForPageLoaded(function() {
+            test.assertTruthy(casper.evaluate(function() {
+                return window.require('core/settings').manifest_url;
+            }), 'Check that manifest_url is defined in settings');            
+        });
+        helpers.done(test);        
+    }
+});
+
+
+casper.test.begin('Test manifest_url', {
+    test: function(test) {
+        helpers.startCasper();
+        helpers.waitForPageLoaded(function() {
+            test.assertTruthy(casper.evaluate(function() {
+                return window.require('core/settings').manifest_url;
+            }), 'Check that manifest_url is defined in settings');            
+        });
+        helpers.done(test);        
+    }
+});
+
+
+casper.test.begin('Test show banner and click download button', {
```

Comment:
you'll also want a "negative" test case to see that the update banner doesn't show up when _setDownloadAvailable is not used